src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -2950,6 +2952,16 @@ def openshift_create(cmd, client, resource_group_name, name,  # pylint: disable=
                 namespace='Microsoft.Network', type='virtualNetwork',
                 name=vnet_peer
             )
+    if workspace_resource_id is not None:
+        workspace_resource_id = workspace_resource_id.strip()
+        if not workspace_resource_id.startswith('/'):
+            workspace_resource_id = '/' + workspace_resource_id
+        if workspace_resource_id.endswith('/'):
+            workspace_resource_id = workspace_resource_id.rstrip('/')
+        monitor_profile = OpenShiftManagedClusterMonitorProfile(enabled=True, workspace_resource_id=workspace_resource_id)
```

Comment:
@troy0820 I don't think it's quite as simple as this.  You can't just graft a struct from the preview API onto the released API - Azure won't recognise it because the request API version will be old.  If loganalytics is requested `az` will need to send a preview API request to the Azure API.  It will also have to handle receiving a preview API response from the Azure API.

If loganalytics is not requested, in some ways it doesn't matter which API version is sent to the Azure API.  It is possibly safer (although more work) to send a request using the released API, however.  Check with MSFT and check to see what the codebase does in this regard for other preview APIs.

I think I'd expect to see:
* if `az` parses server response, it can handle responses in either API version
* `az` creates request using released API, then, if loganalytics is requested, it converts that to the preview API and adds the additional monitorProfile.

Reply:
The way the cli is designed all az openshift calls have to have the same api version which should move to the preview version. The customer can override this value. We are running all tests on the preview version and it is passing.

Reply:
@JackQuincy I don't quite get it - does that mean the PR can't merge until the new API is GA?

Reply:
No we have used preview versions in az in the past. We more have preview features.  But the rest of the api should be unaffected